app/assets/javascripts/views/story_view.js

```diff
@@ -749,6 +750,11 @@ module.exports = FormView.extend({
     this.disableControlButtons();
   },
 
+  attachmentFail: function() {
+    this.uploadInProgress = false;
+    this.disableControlButtons();
+  },
+
   disableControlButtons: function() {
```

Comment:
THis function seems to be a little weird to me, it get some non-direct attribute such `uploadInProgress` to determine if buttons will be enabled or not. More, in attachmentFail function you are intending to enable the buttons, but the `disableControlButtons` is called.

Wasn't better this class be named something like `toggleControlButtons`, receive a boolean state, for example, and use it to determine if buttons should be enable/disabled?